wordpress-mobile/WordPress-Android
# Remove Helpshift dependency and related codes

This PR removes the Helpshift dependency and all the related code for it. The removed code has not been in use in `issue/zendesk-master` branch which this PR targets. I previously attempted to remove Helpshift, but was blocked by our use of it for ipv4 validation in Site Settings. Aside from the removal of Helpshift, the only other change in this PR is to address that by adding a custom regex validation for ipv4 which I got from a [StackOverflow thread](https://stackoverflow.com/a/5667417).

**Notes for reviewing the PR:**
* Some push notification code for Helpshift is removed. I'll implement the Zendesk counterpart of that, which I have already started in #7840. I've left some todos for them, but they will be removed before these changes are merged into `develop`.
* Search for mentions of `Helpshift` in the code. The only ones left should be some analytics code which is already removed in #7904.

**To test:**

There is nothing to test about Helpshift, since it's a simple removal. For the ipv4 validation change:

* Switch to a Jetpack site
* Go to Site Settings -> Jetpack Settings -> Security -> Whitelisted IP addresses
* Try adding valid and invalid ips and make sure validation works correctly.

P.S: The PR is targeting a WIP branch, that's why it doesn't have a milestone.
